staging/src/k8s.io/apiextensions-apiserver/pkg/kcp/cluster_aware_crd_lister.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kcp
+
+import (
+	"context"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// ClusterAwareCRDLister is a CRD lister that is kcp-specific.
+type ClusterAwareCRDLister interface {
+	// List lists all CRDs matching selector.
+	List(ctx context.Context, selector labels.Selector) ([]*v1.CustomResourceDefinition, error)
+	// Get gets a CRD by name.
+	Get(ctx context.Context, name string) (*v1.CustomResourceDefinition, error)
+	// GetLatest gets the current/latest copy of the CRD from the cache. This is necessary to ensure the identity
+	// annotation is present when called by crdHandler.getOrCreateServingInfoFor
```

Comment:
not sure I understand. Also the other two methods return an identity on the CRD, but are informer based? This one is using a client, but does the same thing?

Reply:
None of these uses a client. This is only used by getOrCreateServingInfoFor. The implementation is "refresh latest from cache, ensuring the identity annotation is carried forward since it's not in etcd." (It also add the partial metadata annotation I use to keep track if it's a partial request.)

Reply:
oh, didn't get that on first review. Names matter :)

Reply:
Renamed to Refresh